src/ui/AP2DataPlot2DModel.cc

```diff
@@ -834,3 +881,198 @@ quint64 AP2DataPlot2DModel::getFirstIndex()
 {
     return m_firstIndex;
 }
+
+//******************* Class Error Type **************************
+
+ErrorType::ErrorType() : Timeus(0), SubSys(0), ErrorCode(0)
+{}
+
+bool ErrorType::operator != (const ErrorType &lhs)
+{
+    return ((this->Timeus != lhs.Timeus) || (this->SubSys != lhs.SubSys) || (this->ErrorCode != lhs.ErrorCode));
+}
+
+QString ErrorType::toString()
+{
+    // SubSys ans ErrorCode interpretation was taken from
+    // Ardupilot/ArduCopter/defines.h
+
+    std::stringstream ss;
```

Comment:
Ahh - okay I'm not very familiar with Qt but getting better :-). I'll make a fix for that.